core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -226,10 +226,10 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
               }
             }
           }
-        }
-        lengths[i] = writer.closeAndGetLength();
-        if (file.exists() && !file.delete()) {
-          logger.error("Unable to delete file for partition {}", i);
+          lengths[i] = writer.getNumBytesWritten();
```

Comment:
uhh i think this is still in the try block? should move this outside the try block since we want to get the num-bytes after closing the writer